bind.go

```diff
@@ -35,32 +35,70 @@ func Named(name string, value interface{}) driver.NamedValue {
 	}
 }
 
+var bindNumericRe = regexp.MustCompile(`\$[0-9]+`)
+var bindPositionalRe = regexp.MustCompile(`[?]`)
+
 func bind(tz *time.Location, query string, args ...interface{}) (string, error) {
 	if len(args) == 0 {
 		return query, nil
 	}
 	var (
-		haveNamed   bool
-		haveNumeric bool
+		haveNamed      bool
+		haveNumeric    bool
+		havePositional bool
 	)
+	haveNumeric = bindNumericRe.MatchString(query)
+	havePositional = bindPositionalRe.MatchString(query)
+	if haveNumeric && havePositional {
+		return "", ErrBindMixedParamsFormats
+	}
 	for _, v := range args {
 		switch v.(type) {
 		case driver.NamedValue:
 			haveNamed = true
 		default:
-			haveNumeric = true
 		}
-		if haveNamed && haveNumeric {
-			return "", ErrBindMixedNamedAndNumericParams
+		if haveNamed && (haveNumeric || havePositional) {
+			return "", ErrBindMixedParamsFormats
 		}
 	}
 	if haveNamed {
 		return bindNamed(tz, query, args...)
 	}
-	return bindNumeric(tz, query, args...)
+	if haveNumeric {
+		return bindNumeric(tz, query, args...)
+	}
+	return bindPositional(tz, query, args...)
 }
 
-var bindNumericRe = regexp.MustCompile(`\$[0-9]+`)
+func bindPositional(tz *time.Location, query string, args ...interface{}) (_ string, err error) {
+	var (
+		unbind = make(map[int]struct{})
+		params = make([]string, len(args))
+	)
+	for i, v := range args {
+		if fn, ok := v.(std_driver.Valuer); ok {
+			if v, err = fn.Value(); err != nil {
+				return "", nil
```

Comment:
I don't understand why we return an empty query string and no error here?  Won't that cause weirdness and confusion in the caller?

Reply:
yes this was be consistent with the other methods..i'd need to dig into under what conditions this is used and why we handle it this way. I'll work on a PR to address all 3 methods.